backends/smt2/smtio.py

```diff
@@ -45,6 +45,46 @@ def __init__(self):
         self.asserts = dict()
         self.anyconsts = dict()
 
+    def primary_inputs(self):
+        for name in self.inputs:
+            yield (name, self.wsize[name])
+
+    def clock_inputs(self):
+        for name in self.inputs:
+            if name in ["clk", "clock", "CLK", "CLOCK"]:
+                yield name
+
+    def update(self, stmt):
+        fields = stmt.split()
```

Comment:
The intent behind SmtInfo is that it is a "struct", not a "class". SmtIo does all the SMT-LIB stuff and parsing of yosys-smt2- comments. Imo moving this code from SmtIo to SmtInfo just fractures the code without adding anything to it. It's not like you could use SmtInfo stand-alone without SmtInfo and do anything useful with it.

If anything, SmtInfo should actually be something like a collections.namedtuple. But I don't think that this would actually make the code any cleaner, it would just prevent the next guy from adding methods to it. :)